data_steward/validation/email_notification.py

```diff
@@ -11,17 +11,63 @@
 import mandrill
 from jinja2 import Template
 from matplotlib import image as mpimg
+import requests
 
 # Project imports
 import app_identity
-from constants.utils import bq as bq_consts
+import bq_utils
+from constants.utils import bq as bq_client_consts
+from constants import bq_utils as bq_rest_consts
 from constants.validation import email_notification as consts
 from resources import achilles_images_path
 
 LOGGER = logging.getLogger(__name__)
 
 CONTACT_QUERY_TMPL = Template(consts.CONTACT_LIST_QUERY)
 
+METADATA_URL = 'http://metadata.google.internal/computeMetadata/v1/'
+METADATA_HEADERS = {'Metadata-Flavor': 'Google'}
+SERVICE_ACCOUNT = 'default'
+SCOPES = "https://www.googleapis.com/auth/drive,https://www.googleapis.com/auth/cloud-platform"
+
+
+def get_access_token():
+    url = f'{METADATA_URL}instance/service-accounts/{SERVICE_ACCOUNT}/token?scopes={SCOPES}'
+    # Request an access token from the metadata server.
+    r = requests.get(url, headers=METADATA_HEADERS)
+    r.raise_for_status()
+    # Extract the access token from the response.
+    access_token = r.json()['access_token']
+    return access_token
+
+
+def post_query(project_id):
+    url = f'https://www.googleapis.com/bigquery/v2/projects/{project_id}/queries'
```

Comment:
It's possible someone else may query either the current or some other drive-backed table and run into the same scope issue. We should aim to prevent their having to reinvent this solution.

At minimum I recommend refactoring this by 
1. relocating it to `utils.bq`
1. having it accept the parameters `query: str` and `external_data_scopes:List[str]`
1. adding a docstring 😄 

Please give due diligence (<1 hour) to possibly injecting the token into the client somehow. I'll let you know if I find anything.

Reply:
Done! Unfortunately, I couldn't find any info on using access tokens for the client library. They really want everything done under the hood and require a credentials object instead to determine the access tokens.